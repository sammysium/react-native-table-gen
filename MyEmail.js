import PropTypes from 'prop-types';
import React from 'react';
import { StyleSheet, Text, Linking } from 'react-native';

const MyComponent = ({
  email,
  styles
}) => {
  const txtStyle = styles !== undefined ? styles : defaultStyles.text;

  return (
    <>
      <Text style={txtStyle} onPress={() => Linking.openURL(`mailto:${email}`)}>{email}</Text>
    </>
  );
};

const defaultStyles = StyleSheet.create({
  text: {
    color: '#00f',
    marginBottom: 4,
    marginRight: 10
  }
});
MyComponent.propTypes = {
  email: PropTypes.string.isRequired,
  styles: PropTypes.instanceOf(Object)
};
MyComponent.defaultProps ={
  styles: undefined
};
const MyEmail = React.memo(MyComponent);
export default MyEmail;
