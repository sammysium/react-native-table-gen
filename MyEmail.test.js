import React from 'react';
import { Text, Linking } from 'react-native';
import { render, fireEvent } from 'react-native-testing-library';
import MyEmail from './MyEmail';

describe('component MyEmail', () => {
  const email = 'email@example.com';
  it('renders correctly', () => {
    const snapShotTree = render(<MyEmail email={email} />).toJSON();
    expect(snapShotTree).toMatchSnapshot();
  });

  it('should display the passed email', () => {
    const App = () => (
      <>
        <MyEmail email={email} />
      </>
    );
    const wrapper = render(<App />);
    const label = wrapper.UNSAFE_getByType(Text);
    expect(label.props.children).toStrictEqual(email);
  });

  it('should open email client', () => {
    const App = () => (
      <>
        <MyEmail email={email} />
      </>
    );

    const spy = jest.spyOn(Linking,'openURL');

    const wrapper = render(<App />);
    const label = wrapper.UNSAFE_getByType(Text);
    fireEvent.press(label);
    expect(spy).toHaveBeenCalledWith(`mailto:${email}`)
  });
});
