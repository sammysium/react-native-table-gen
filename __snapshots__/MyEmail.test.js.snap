// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component MyEmail renders correctly 1`] = `
<Text
  onPress={[Function]}
  style={
    Object {
      "color": "#00f",
      "marginBottom": 4,
      "marginRight": 10,
    }
  }
>
  email@example.com
</Text>
`;
